File: jvm/core/src/main/java/se/idkollen/client/Environment.java
package se.idkollen.client;

/** Pre-configured API base URL selection. */
public enum Environment {
    /** Production environment: {@code https://api.idkollen.se}. */
    PRODUCTION("https://api.idkollen.se"),
    /** Staging environment: {@code https://stgapi.idkollen.se}. */
    STAGING("https://stgapi.idkollen.se");

    public final String baseUrl;

    Environment(String baseUrl) {
        this.baseUrl = baseUrl;
    }
}

File: ruby/lib/idkollen/environment.rb
# frozen_string_literal: true

module Idkollen
  module Environment
    PRODUCTION = :production
    STAGING    = :staging

    BASE_URLS = {
      PRODUCTION => 'https://api.idkollen.se',
      STAGING    => 'https://stgapi.idkollen.se',
    }.freeze

    def self.base_url(env)
      BASE_URLS.fetch(env)
    end
  end
end
